assembly/src/assembler/module_graph/mod.rs

```diff
@@ -376,26 +514,41 @@ impl ModuleGraph {
             Entry::Occupied(ref mut entry) => {
                 let prev_id = entry.get()[0];
                 if prev_id != id {
-                    // Multiple procedures with the same root, but incompatible
-                    let prev = &self.modules[prev_id.module.as_usize()][prev_id.index];
-                    let current = &self.modules[id.module.as_usize()][id.index];
-                    if prev.num_locals() != current.num_locals() {
-                        let prev_module = self.modules[prev_id.module.as_usize()].path();
-                        let prev_name = FullyQualifiedProcedureName {
-                            span: prev.span(),
-                            module: prev_module.clone(),
-                            name: prev.name().clone(),
-                        };
-                        let current_module = self.modules[id.module.as_usize()].path();
-                        let current_name = FullyQualifiedProcedureName {
-                            span: current.span(),
-                            module: current_module.clone(),
-                            name: current.name().clone(),
-                        };
-                        return Err(AssemblyError::ConflictingDefinitions {
-                            first: prev_name,
-                            second: current_name,
-                        });
+                    let prev_proc = {
+                        match &self.modules[prev_id.module.as_usize()] {
+                            WrappedModule::Ast(module) => Some(&module[prev_id.index]),
+                            WrappedModule::Info(_) => None,
+                        }
+                    };
+                    let current_proc = {
+                        match &self.modules[id.module.as_usize()] {
+                            WrappedModule::Ast(module) => Some(&module[id.index]),
+                            WrappedModule::Info(_) => None,
+                        }
+                    };
+
+                    // Note: For compiled procedures, we can't check further if they're compatible,
+                    // so we assume they are.
```

Comment:
Note for the future: if two procedures have the same MAST root, they would always have the same number of locals (because changing the number of locals would change the MAST root). So, if the number of locals differs somehow between two procedures with the same MAST root, one (or both) of them have incorrect number of locals in the metadata.